packages/components/nodes/moderation/SimplePromptModeration/SimplePromptModerationRunner.ts

```diff
@@ -1,23 +1,38 @@
 import { Moderation } from '../Moderation'
+import { BaseChatModel } from 'langchain/chat_models/base'
 
 export class SimplePromptModerationRunner implements Moderation {
     private readonly denyList: string = ''
     private readonly moderationErrorMessage: string = ''
+    private readonly model: BaseChatModel
 
-    constructor(denyList: string, moderationErrorMessage: string) {
+    constructor(denyList: string, moderationErrorMessage: string, model?: BaseChatModel) {
         this.denyList = denyList
         if (denyList.indexOf('\n') === -1) {
             this.denyList += '\n'
         }
         this.moderationErrorMessage = moderationErrorMessage
+        if (model) this.model = model
     }
 
     async checkForViolations(input: string): Promise<string> {
-        this.denyList.split('\n').forEach((denyListItem) => {
-            if (denyListItem && denyListItem !== '' && input.toLowerCase().includes(denyListItem.toLowerCase())) {
-                throw Error(this.moderationErrorMessage)
+        if (this.model) {
+            const denyArray = this.denyList.split('\n')
+            for (const denyStr of denyArray) {
+                const res = await this.model.invoke(
+                    `Are these two sentences similar to each other? Only return Yes or No.\nFirst sentence: ${input}\nSecond sentence: ${denyStr}`
```

Comment:
can we not send the entire denyList and make one LLM call with an adjusted prompt?

Reply:
we can add an option to let user select if they want to make one LLM call per deny string, or all together at once

Reply:
been trying to make a call like this: 
`const res = await this.model.invoke(
                    `Is this sentence: "${input}" similar to any of the sentences below? Only return Yes or No.\nSentences: ${this.denyList}`
                )`
However it always return false..

Reply:
will let it run one LLM call per deny for now, as i cant get consistent performance when using the whole list